crates/red_knot_python_semantic/src/semantic_index/ast_ids.rs

```diff
@@ -30,24 +26,13 @@ use crate::Db;
 /// ```
 pub(crate) struct AstIds {
     /// Maps expression ids to their expressions.
-    expressions: IndexVec<ScopedExpressionId, AstNodeRef<ast::Expr>>,
+    expressions: IndexVec<ScopedExpressionId, ()>,
```

Comment:
We never lookup the expression node by an id. All we need is a unique ID. We may want to reconsider this more holistically, but I would rather not do it as part of this PR.